schema/parser.cpp

```diff
@@ -514,13 +514,35 @@ Element Parser::parseElement(ParserContext *context, const QDomElement &element,
             // qDebug() << "childName:" << childName.localName();
             if (childName.localName() == QLatin1String("complexType")) {
                 ComplexType ct = parseComplexType(context, childElement);
-
-                ct.setName(newElement.name());
                 ct.setAnonymous(true);
-                d->mComplexTypes.append(ct);
-                qCDebug(parser)
-                        << " found nested complexType element, type name is now element name, i.e. "
-                        << ct.name() << "newElement.setType" << ct.qualifiedName();
+                ct.setName(newElement.name());
+                int i = 0;
+                bool typeExists = false;
+                ComplexType existingType =
+                        d->mComplexTypes.complexType(QName(ct.nameSpace(), ct.name()));
+                while (existingType != ComplexType()) {
```

Comment:
```suggestion
                while (existingType.isNull()) {
```

(This calls XmlElement::isNull())

Creating an instance just to compare with it is a bit wasteful.

Reply:
Just testing with isNull() method doesn't work.
Do avoid creating an empty/null object we can:
```
ComplexType nullType;
while (existingType != nullType) {
```
or update isNull method but I don't know about the possible side effects.

Reply:
Can you tell me more about why `isNull` doesn't work? Is it never true? Or is it true when it shouldn't be?

`ComplexTypeList::complexType()` returns `ComplexType()` if not found, which ends up calling the `XmlElement()` default constructor, which leaves `mName` empty, so `isNull()` returns true. So surely it should manage to return true when complexType() can't find the type. Is the issue that is also returns true in cases where complexType found a type, but it has no name? Since qualifiedName() includes mName, that would only be true if ct.name() is empty? Can this really happen?

Reply:
I just just copy/past what suggesting without noticing the missing ! bit at the start.
No wonder the loop... looped for ever...